template/gitlab-runner.tftpl

```diff
@@ -39,17 +39,48 @@ then
     gitlab_runner_registration_token=$(aws ssm get-parameter --name "${secure_parameter_store_gitlab_runner_registration_token_name}" --with-decryption --region "${secure_parameter_store_region}" | jq -r ".Parameter | .Value")
 fi
 
+# fetch gitlab token from SSM
+gitlab_token=$(aws ssm get-parameter --name "${secure_parameter_store_gitlab_token_name}" --with-decryption --region "${secure_parameter_store_region}" | jq -r ".Parameter | .Value")
+
 if [[ "${runners_token}" == "__REPLACED_BY_USER_DATA__" && "$token" == "null" ]] || [[ "$valid_token" == "false" ]]
 then
-  token=$(curl ${curl_cacert} --request POST -L "${runners_gitlab_url}/api/v4/runners" \
-    --form "token=$gitlab_runner_registration_token" \
-    --form "tag_list=${gitlab_runner_tag_list}" \
-    --form "description=${gitlab_runner_description}" \
-    --form "locked=${gitlab_runner_locked_to_project}" \
-    --form "run_untagged=${gitlab_runner_run_untagged}" \
-    --form "maximum_timeout=${gitlab_runner_maximum_timeout}" \
-    --form "access_level=${gitlab_runner_access_level}" \
-    | jq -r .token)
+  if [ "${use_new_runner_authentication_gitlab_16}" == "false" ]
+  then
+    token=$(curl ${curl_cacert} --request POST -L "${runners_gitlab_url}/api/v4/runners" \
+      --form "token=$gitlab_runner_registration_token" \
+      --form "tag_list=${gitlab_runner_tag_list}" \
+      --form "description=${gitlab_runner_description}" \
+      --form "locked=${gitlab_runner_locked_to_project}" \
+      --form "run_untagged=${gitlab_runner_run_untagged}" \
+      --form "maximum_timeout=${gitlab_runner_maximum_timeout}" \
+      --form "access_level=${gitlab_runner_access_level}" \
+      | jq -r .token)
+  else
+    runner_type_param=""
+    if [ "${gitlab_runner_type}" = "group_type" ]; then
+      if [ ! -z "${gitlab_runner_group_id}" ]; then
+        echo "If the runner type is group_type, you must specify a group_id".
+      fi
+      runner_type_param='--form group_id=${gitlab_runner_group_id}'
+    elif [ "${gitlab_runner_type}" = "project_type" ]; then
+      if [ ! -z "${gitlab_runner_project_id}" ]; then
+        echo "If the runner type is project_type, you must specify a project_id".
```

Comment:
suggestion: raise an exit state with `exit`. The GitLab Runner is not able to work. Same as above.